### ISSUE TITLE
Compiling from cargo: not found in the crate root

### ISSUE THREAD

**Issue by madskjeldgaard** [tramhao/termusic#91]
Hi! Trying to install termusic on an Apple Sillicon mac using cargo, but I get the following errors when running `cargo install termusic`:

```
 cargo install termusic
    Updating crates.io index
  Installing termusic v0.7.3
  Downloaded console v0.15.2
  Downloaded unicode-linebreak v0.1.4
  Downloaded itoa v1.0.4
  Downloaded image v0.24.4
  Downloaded crossbeam-epoch v0.9.11
  Downloaded exr v1.5.2
  Downloaded jobserver v0.1.25
  Downloaded miniz_oxide v0.6.2
  Downloaded half v2.1.0
  Downloaded syn v1.0.102
  Downloaded webpki-roots v0.22.5
  Downloaded unicode-ident v1.0.5
  Downloaded thiserror-impl v1.0.37
  Downloaded serde_json v1.0.86
  Downloaded smallvec v1.10.0
  Downloaded proc-macro2 v1.0.46
  Downloaded libc v0.2.135
  Downloaded thiserror v1.0.37
  Downloaded crossbeam-utils v0.8.12
  Downloaded 19 crates (2.2 MB) in 7.88s
   Compiling cfg-if v1.0.0
   Compiling libc v0.2.135
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.46
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.5
   Compiling log v0.4.17
   Compiling memchr v2.5.0
   Compiling bitflags v1.3.2
   Compiling syn v1.0.102
   Compiling scopeguard v1.1.0
   Compiling bytemuck v1.12.1
   Compiling lazy_static v1.4.0
   Compiling regex-syntax v0.6.27
   Compiling arrayvec v0.7.2
   Compiling version_check v0.9.4
   Compiling cc v1.0.73
   Compiling encoding_rs v0.8.31
   Compiling smallvec v1.10.0
   Compiling crossbeam-utils v0.8.12
   Compiling adler v1.0.2
   Compiling crc32fast v1.3.2
   Compiling parking_lot_core v0.9.3
   Compiling signal-hook v0.3.14
   Compiling once_cell v1.15.0
   Compiling glob v0.3.0
   Compiling rayon-core v1.9.3
   Compiling serde_derive v1.0.145
   Compiling minimal-lexical v0.2.1
   Compiling serde v1.0.145
   Compiling futures-core v0.3.24
   Compiling thiserror v1.0.37
   Compiling untrusted v0.7.1
   Compiling bindgen v0.59.2
   Compiling unicode-width v0.1.10
   Compiling shlex v1.1.0
   Compiling either v1.8.0
   Compiling tinyvec_macros v0.1.0
   Compiling peeking_take_while v0.1.2
   Compiling rustc-hash v1.1.0
   Compiling lazycell v1.3.0
   Compiling byteorder v1.4.3
   Compiling weezl v0.1.7
   Compiling pkg-config v0.3.25
   Compiling futures-sink v0.3.24
   Compiling cassowary v0.3.0
   Compiling unicode-segmentation v1.10.0
   Compiling vcpkg v0.2.15
   Compiling bit_field v0.10.1
   Compiling lebe v0.5.2
   Compiling rustls v0.20.6
   Compiling color_quant v1.1.0
   Compiling percent-encoding v2.2.0
   Compiling unicode-bidi v0.3.8
   Compiling base64 v0.13.0
   Compiling half v2.1.0
   Compiling fastrand v1.8.0
   Compiling smawk v0.3.1
   Compiling serde_json v1.0.86
   Compiling remove_dir_all v0.5.3
   Compiling core-foundation-sys v0.8.3
   Compiling ppv-lite86 v0.2.16
   Compiling pinyin v0.9.0
   Compiling anyhow v1.0.65
   Compiling scoped_threadpool v0.1.9
   Compiling same-file v1.0.6
   Compiling orange-trees v0.1.0
   Compiling linked-hash-map v0.5.6
   Compiling fallible-iterator v0.2.0
   Compiling paste v1.0.9
   Compiling itoa v1.0.4
   Compiling termcolor v1.1.3
   Compiling ryu v1.0.11
   Compiling fallible-streaming-iterator v0.1.9
   Compiling chunked_transfer v1.4.0
   Compiling urlencoding v2.1.2
   Compiling wildmatch v2.1.1
   Compiling hex v0.4.3
   Compiling libaes v0.6.4
   Compiling md5 v0.7.0
   Compiling lexopt v0.2.1
   Compiling rgb v0.8.34
   Compiling lock_api v0.4.9
   Compiling memoffset v0.6.5
   Compiling crossbeam-epoch v0.9.11
   Compiling num-traits v0.2.15
   Compiling rayon v1.5.3
   Compiling num-integer v0.1.45
   Compiling num-rational v0.4.1
   Compiling num-bigint v0.4.3
   Compiling libloading v0.7.3
   Compiling ahash v0.7.6
   Compiling ring v0.16.20
   Compiling miniz_oxide v0.5.4
   Compiling miniz_oxide v0.6.2
   Compiling clang-sys v1.4.0
   Compiling tinyvec v1.6.0
   Compiling ogg_pager v0.3.2
   Compiling form_urlencoded v1.1.0
   Compiling gif v0.11.4
   Compiling libsqlite3-sys v0.25.1
   Compiling walkdir v2.3.2
   Compiling yaml-rust v0.4.5
   Compiling ansi_colours v1.1.1
   Compiling unicode-normalization v0.1.22
   Compiling symphonia-core v0.5.1
   Compiling aho-corasick v0.7.19
   Compiling nom v7.1.1
   Compiling getrandom v0.2.7
   Compiling quick-xml v0.23.1
   Compiling crossbeam-channel v0.5.6
   Compiling signal-hook-registry v1.4.0
   Compiling num_cpus v1.13.1
   Compiling mio v0.8.4
   Compiling terminal_size v0.1.17
   Compiling dirs-sys v0.3.7
   Compiling tempfile v3.3.0
   Compiling mach v0.3.2
   Compiling flate2 v1.0.24
   Compiling idna v0.3.0
   Compiling spin v0.9.4
   Compiling regex v1.6.0
   Compiling symphonia-metadata v0.5.1
   Compiling symphonia-codec-alac v0.5.1
   Compiling symphonia-codec-pcm v0.5.1
   Compiling symphonia-codec-aac v0.5.1
   Compiling include_dir_macros v0.7.2
   Compiling cexpr v0.6.0
   Compiling nanorand v0.7.0
   Compiling rand_core v0.6.4
   Compiling threadpool v1.8.1
   Compiling parking_lot v0.12.1
   Compiling console v0.15.2
   Compiling dirs v4.0.0
   Compiling png v0.17.6
   Compiling id3 v1.3.0
   Compiling url v2.3.1
   Compiling crossbeam-deque v0.8.2
   Compiling hashbrown v0.12.3
   Compiling symphonia-utils-xiph v0.5.1
   Compiling symphonia-bundle-mp3 v0.5.1
   Compiling symphonia-format-wav v0.5.1
   Compiling include_dir v0.7.2
   Compiling rand_chacha v0.3.1
   Compiling signal-hook-mio v0.2.3
   Compiling webpki v0.22.0
   Compiling sct v0.7.0
   Compiling shellexpand v2.1.2
   Compiling thiserror-impl v1.0.37
   Compiling pin-project-internal v1.0.12
   Compiling lazy-regex-proc_macros v2.3.0
   Compiling tuirealm_derive v1.0.0
   Compiling lofty_attr v0.3.0
   Compiling symphonia-format-ogg v0.5.1
   Compiling symphonia-codec-vorbis v0.5.1
   Compiling symphonia-bundle-flac v0.5.1
   Compiling symphonia-format-mkv v0.5.1
   Compiling symphonia-format-isomp4 v0.5.1
   Compiling unicode-linebreak v0.1.4
   Compiling hashlink v0.8.1
   Compiling rand v0.8.5
   Compiling crossterm v0.25.0
   Compiling crossterm v0.23.2
   Compiling webpki-roots v0.22.5
   Compiling pin-project v1.0.12
   Compiling lazy-regex v2.3.0
   Compiling lofty v0.8.1
   Compiling coreaudio-sys v0.2.10
   Compiling symphonia v0.5.1
   Compiling tui v0.19.0
   Compiling ytd-rs v0.1.7
   Compiling flume v0.10.14
   Compiling ureq v2.5.0
   Compiling jpeg-decoder v0.2.6
   Compiling exr v1.5.2
   Compiling tiff v0.7.3
   Compiling tuirealm v1.8.0
   Compiling toml v0.5.9
   Compiling textwrap v0.15.1
   Compiling image v0.24.4
   Compiling tui-realm-treeview v1.1.0
   Compiling tui-realm-stdlib v1.1.7
   Compiling rusqlite v0.28.0
   Compiling viuer v0.6.1
   Compiling coreaudio-rs v0.10.0
   Compiling termusic v0.7.3
error[E0432]: unresolved import `crate::DefaultStreamConfigError`
 --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:8:5
  |
8 | use crate::DefaultStreamConfigError;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `DefaultStreamConfigError` in the root

error[E0432]: unresolved imports `crate::BuildStreamError`, `crate::SupportedStreamConfigsError`
 --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:9:13
  |
9 | use crate::{BuildStreamError, SupportedStreamConfigsError};
  |             ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `SupportedStreamConfigsError` in the root
  |             |
  |             no `BuildStreamError` in the root

error[E0432]: unresolved imports `crate::BackendSpecificError`, `crate::SampleFormat`, `crate::StreamConfig`
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:11:13
   |
11 | use crate::{BackendSpecificError, SampleFormat, StreamConfig};
   |             ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^ no `StreamConfig` in the root
   |             |                     |
   |             |                     no `SampleFormat` in the root
   |             no `BackendSpecificError` in the root

error[E0432]: unresolved import `crate::traits`
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:23:12
   |
23 | use crate::traits::{DeviceTrait, HostTrait, StreamTrait};
   |            ^^^^^^
   |            |
   |            unresolved import
   |            help: a similar path exists: `crate::player::rusty_backend::cpal::traits`

error[E0432]: unresolved imports `crate::BackendSpecificError`, `crate::BufferSize`, `crate::BuildStreamError`, `crate::ChannelCount`, `crate::Data`, `crate::DefaultStreamConfigError`, `crate::DeviceNameError`, `crate::DevicesError`, `crate::InputCallbackInfo`, `crate::OutputCallbackInfo`, `crate::PauseStreamError`, `crate::PlayStreamError`, `crate::SampleFormat`, `crate::SampleRate`, `crate::StreamConfig`, `crate::StreamError`, `crate::SupportedBufferSize`, `crate::SupportedStreamConfig`, `crate::SupportedStreamConfigRange`, `crate::SupportedStreamConfigsError`
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:25:5
   |
25 |     BackendSpecificError, BufferSize, BuildStreamError, ChannelCount, Data,
   |     ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^  ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^ no `Data` in the root
   |     |                     |           |                 |
   |     |                     |           |                 no `ChannelCount` in the root
   |     |                     |           no `BuildStreamError` in the root
   |     |                     no `BufferSize` in the root
   |     no `BackendSpecificError` in the root
26 |     DefaultStreamConfigError, DeviceNameError, DevicesError, InputCallbackInfo, OutputCallbackInfo,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^ no `OutputCallbackInfo` in the root
   |     |                         |                |             |
   |     |                         |                |             no `InputCallbackInfo` in the root
   |     |                         |                no `DevicesError` in the root
   |     |                         no `DeviceNameError` in the root
   |     no `DefaultStreamConfigError` in the root
27 |     PauseStreamError, PlayStreamError, SampleFormat, SampleRate, StreamConfig, StreamError,
   |     ^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^
28 |     SupportedBufferSize, SupportedStreamConfig, SupportedStreamConfigRange,
   |     ^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^
29 |     SupportedStreamConfigsError,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0432]: unresolved imports `crate::BackendSpecificError`, `crate::DevicesError`, `crate::SupportedStreamConfigRange`
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/enumerate.rs:11:13
   |
11 | use crate::{BackendSpecificError, DevicesError, SupportedStreamConfigRange};
   |             ^^^^^^^^^^^^^^^^^^^^  ^^^^^^^^^^^^  ^^^^^^^^^^^^^^^^^^^^^^^^^^ no `SupportedStreamConfigRange` in the root
   |             |                     |
   |             |                     no `DevicesError` in the root
   |             no `BackendSpecificError` in the root

error[E0412]: cannot find type `HostUnavailable` in the crate root
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:53:41
   |
53 |     pub fn new() -> Result<Self, crate::HostUnavailable> {
   |                                         ^^^^^^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
4  | use crate::player::rusty_backend::cpal::HostUnavailable;
   |
help: if you import `HostUnavailable`, refer to it directly
   |
53 -     pub fn new() -> Result<Self, crate::HostUnavailable> {
53 +     pub fn new() -> Result<Self, HostUnavailable> {
   |

error[E0422]: cannot find struct, variant or union type `InputStreamTimestamp` in the crate root
   --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:683:36
    |
683 |             let timestamp = crate::InputStreamTimestamp { callback, capture };
    |                                    ^^^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
help: consider importing this struct
    |
4   | use crate::player::rusty_backend::cpal::InputStreamTimestamp;
    |
help: if you import `InputStreamTimestamp`, refer to it directly
    |
683 -             let timestamp = crate::InputStreamTimestamp { callback, capture };
683 +             let timestamp = InputStreamTimestamp { callback, capture };
    |

error[E0422]: cannot find struct, variant or union type `OutputStreamTimestamp` in the crate root
   --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/macos/mod.rs:775:36
    |
775 |             let timestamp = crate::OutputStreamTimestamp { callback, playback };
    |                                    ^^^^^^^^^^^^^^^^^^^^^ not found in the crate root
    |
help: consider importing this struct
    |
4   | use crate::player::rusty_backend::cpal::OutputStreamTimestamp;
    |
help: if you import `OutputStreamTimestamp`, refer to it directly
    |
775 -             let timestamp = crate::OutputStreamTimestamp { callback, playback };
775 +             let timestamp = OutputStreamTimestamp { callback, playback };
    |

error[E0412]: cannot find type `StreamInstant` in the crate root
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:73:20
   |
73 | ) -> Result<crate::StreamInstant, BackendSpecificError> {
   |                    ^^^^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
3  | use crate::player::rusty_backend::cpal::StreamInstant;
   |
help: if you import `StreamInstant`, refer to it directly
   |
73 - ) -> Result<crate::StreamInstant, BackendSpecificError> {
73 + ) -> Result<StreamInstant, BackendSpecificError> {
   |

error[E0433]: failed to resolve: could not find `StreamInstant` in the crate root
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:80:15
   |
80 |     Ok(crate::StreamInstant::new(secs as i64, subsec_nanos as u32))
   |               ^^^^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
3  | use crate::player::rusty_backend::cpal::StreamInstant;
   |
help: if you import `StreamInstant`, refer to it directly
   |
80 -     Ok(crate::StreamInstant::new(secs as i64, subsec_nanos as u32))
80 +     Ok(StreamInstant::new(secs as i64, subsec_nanos as u32))
   |

error[E0412]: cannot find type `SampleRate` in the crate root
  --> /Users/mads/.cargo/registry/src/github.com-1ecc6299db9ec823/termusic-0.7.3/src/player/rusty_backend/cpal/host/coreaudio/mod.rs:84:51
   |
84 | fn frames_to_duration(frames: usize, rate: crate::SampleRate) -> std::time::Duration {
   |                                                   ^^^^^^^^^^ not found in the crate root
   |
help: consider importing this struct
   |
3  | use crate::player::rusty_backend::cpal::SampleRate;
   |
help: if you import `SampleRate`, refer to it directly
   |
84 - fn frames_to_duration(frames: usize, rate: crate::SampleRate) -> std::time::Duration {
84 + fn frames_to_duration(frames: usize, rate: SampleRate) -> std::time::Duration {
   |

Some errors have detailed explanations: E0412, E0422, E0432, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `termusic` due to 12 previous errors
error: failed to compile `termusic v0.7.3`, intermediate artifacts can be found at `/var/folders/qz/2bw2g_5s1c399xtb2xtktgkr0000gn/T/cargo-installnQNeA9`

```

**Comment by tramhao** [comment]
This is the same with issue #90 . Please checkout latest master code to compile. If it works, I'll release it as v0.7.4. Thanks.

**Comment by madskjeldgaard** [comment]
> released in v0.7.4

Thanks a lot. It works with this update
